tests/test_filter.py

```diff
@@ -120,184 +120,14 @@ def test_filter_spec(filter_args, expected_filter_spec):
-def vector_grid_type_and_input_ds(request):
-    grid_type = request.param
-    ny, nx = (128, 256)
-
-    grid_vars = {}
-    if grid_type == GridType.VECTOR_C_GRID:
-        # construct spherical coordinate system similar to MOM6 NeverWorld2 grid
-        # define latitudes and longitudes
-        lat_min = -70
-        lat_max = 70
-        lat_u = np.linspace(
-            lat_min + 0.5 * (lat_max - lat_min) / ny,
-            lat_max - 0.5 * (lat_max - lat_min) / ny,
-            ny,
-        )
-        lat_v = np.linspace(lat_min + (lat_max - lat_min) / ny, lat_max, ny)
-        lon_min = 0
-        lon_max = 60
-        lon_u = np.linspace(lon_min + (lon_max - lon_min) / nx, lon_max, nx)
-        lon_v = np.linspace(
-            lon_min + 0.5 * (lon_max - lon_min) / nx,
-            lon_max - 0.5 * (lon_max - lon_min) / nx,
-            nx,
-        )
-        (geolon_u, geolat_u) = np.meshgrid(lon_u, lat_u)
-        (geolon_v, geolat_v) = np.meshgrid(lon_v, lat_v)
-        # radius of a random planet smaller than Earth
-        R = 6378000 * np.random.rand(1)
-        # dx varies spatially
-        dxCu = R * np.cos(geolat_u / 360 * 2 * np.pi)
-        dxCv = R * np.cos(geolat_v / 360 * 2 * np.pi)
-        dxBu = dxCv + np.roll(dxCv, -1, axis=1)
-        dxT = dxCu + np.roll(dxCu, 1, axis=1)
-        da_dxCu = xr.DataArray(dxCu, dims=["y", "x"])
-        da_dxCv = xr.DataArray(dxCv, dims=["y", "x"])
-        da_dxBu = xr.DataArray(dxBu, dims=["y", "x"])
-        da_dxT = xr.DataArray(dxT, dims=["y", "x"])
-        # dy is set constant, equal to dx at the equator
-        dy = np.max(dxCu) * np.ones((ny, nx))
-        da_dy = xr.DataArray(dy, dims=["y", "x"])
-        # compute grid cell areas
-        area_u = dxCu * dy
-        area_v = dxCv * dy
-        da_area_u = xr.DataArray(area_u, dims=["y", "x"])
-        da_area_v = xr.DataArray(area_v, dims=["y", "x"])
-        # set isotropic and anisotropic kappas
-        kappa_data = np.ones((ny, nx))
-        da_kappa = xr.DataArray(kappa_data, dims=["y", "x"])
-        # put a big island in the middle
-        mask_data = np.ones((ny, nx))
-        mask_data[: (ny // 2), : (nx // 2)] = 0
-        da_mask = xr.DataArray(mask_data, dims=["y", "x"])
-        grid_vars = {
-            "wet_mask_t": da_mask,
-            "wet_mask_q": da_mask,
-            "dxT": da_dxT,
-            "dyT": da_dy,
-            "dxCu": da_dxCu,
-            "dyCu": da_dy,
-            "dxCv": da_dxCv,
-            "dyCv": da_dy,
-            "dxBu": da_dxBu,
-            "dyBu": da_dy,
-            "area_u": da_area_u,
-            "area_v": da_area_v,
-            "kappa_iso": da_kappa,
-            "kappa_aniso": da_kappa,
-        }
-    data_u = np.random.rand(ny, nx)
-    data_v = np.random.rand(ny, nx)
-    da_u = xr.DataArray(data_u, dims=["y", "x"])
-    da_v = xr.DataArray(data_v, dims=["y", "x"])
-
-    return grid_type, da_u, da_v, grid_vars, geolat_u
-
```

Comment:
Can you just help me understand why all this code is being removed? Is it redundant with the code in conftest.py?

Reply:
Yes, exactly. The only two differences between the removed code and the code in `conftest.py` were:

1. `conftest.py` created numpy array test data (specific to `test_kernels.py`), whereas the removed code created xarray DataArray test data (specific to `test_filter.py`). The new fixtures `grid_type_and_input_ds` and `vector_grid_type_and_input_ds` in `conftest.py` take care of the numpy array--> xarray DataArray conversion, but without all the redundancy.
2.  The random number generator for the test data. Using the explicit constant seed for the random number generator in `conftest.py` ensures that we are always testing on the same data - a pre-requisite for the filter verification tests, exactly as done in #79.